include/pybind11/pybind11.h

```diff
@@ -48,6 +45,20 @@
 #  include <cxxabi.h>
 #endif
 
+/* https://stackoverflow.com/questions/46798456/handling-gccs-noexcept-type-warning
+   This warning is about ABI compatibility, not code health.
+   It is only actually needed in a couple places, but apparently GCC 7 "generates this warning if
+   and only if the first template instantiation ... involves noexcept" [stackoverflow], therefore
+   it could get triggered from seemingly random places, depending on user code.
+   It seems very unlikely that this warning will be useful to anyone (no other GCC version
+   generates it), but to be maximally accommodating we turn it off here instead of using a
+   command line option.
```

Comment:
>  other GCC version generates it
🤣 

```suggestion
   No other GCC version generates this warning.
```

Too much information, we want pybind11 to compile out of the box with no warnings if possible, and that's why we carefully remove a few warnings, including this one.

Reply:
Change applied. Thanks!